src/bindings/python/tests/test_graph/test_preprocess.py

```diff
@@ -728,3 +728,71 @@ def test_graph_set_layout_by_layout_class_thow_exception():
         layout = Layout("1-2-3D")
         ppp.input().model().set_layout(layout)
     assert "Layout name is invalid" in str(e.value)
+
+
+@pytest.mark.parametrize(
+    ("pads_begin", "pads_end", "values", "mode"),
+    [([0, 0, 0, 0], [0, 0, 1, 1], 0, PaddingMode.CONSTANT)])
+def test_pad_vector_constant_layout(pads_begin, pads_end, values, mode):
+    shape = [1, 3, 200, 200]
+    parameter_a = ops.parameter(shape, dtype=np.float32, name="RGB_input")
+    model = parameter_a
+    model = Model(model, [parameter_a], "TestModel")
+    ppp = PrePostProcessor(model)
+    ppp.input().tensor().set_shape([1, 3, 199, 199])
+    ppp.input().preprocess().pad(pads_begin, pads_end, values, mode)
+    assert ppp.build()
+    assert list(model.get_output_shape(0)) == shape
+
+
+@pytest.mark.parametrize(
+    ("pads_begin", "pads_end", "values", "mode"),
+    [([0, 0, -2, 0], [0, 0, -4, 1], 0, PaddingMode.CONSTANT)]
+)
+def test_pad_vector_out_of_range(pads_begin, pads_end, values, mode):
+    shape = [1, 3, 5, 5]
+    parameter_a = ops.parameter(shape, dtype=np.float32, name="A")
+    model = parameter_a
+    model = Model(model, [parameter_a], "TestModel")
+    ppp = PrePostProcessor(model)
+    with pytest.raises(RuntimeError) as e:
+        ppp.input().preprocess().pad(pads_begin, pads_end, values, mode)
+        ppp.build()
+    assert "not aligned with original parameter's shape" in str(e.value)
+    assert list(model.get_output_shape(0)) == shape
+
+
+@pytest.mark.parametrize(
+    ("pads_begin", "pads_end", "values", "mode"),
+    [([0, 0, 2, 0, 1], [0, 0, 4, 1, 1], 0, PaddingMode.CONSTANT)]
+)
+def test_pad_vector_dim_mismatch(pads_begin, pads_end, values, mode):
+    shape = [1, 3, 5, 5]
+    parameter_a = ops.parameter(shape, dtype=np.float32, name="A")
+    model = parameter_a
+    model = Model(model, [parameter_a], "TestModel")
+    ppp = PrePostProcessor(model)
+    with pytest.raises(RuntimeError) as e:
+        ppp.input().preprocess().pad(pads_begin, pads_end, values, mode)
+        ppp.build()
+    assert "mismatches with rank of input" in str(e.value)
+    assert list(model.get_output_shape(0)) == shape
```

Comment:
We should check `ppp.get_output_shape(0))` rather than `model.get_output_shape(0)`